src/Microsoft.DotNet.Build.CloudTestTasks/FilterForUpload.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.DotNet.Build.CloudTestTasks
+{
+    public sealed class FilterForUpload : Task
+    {
+        [Required]
+        public string TestOutputDirListsLocation { get; set; }
+
+
+        [Required]
+        public string FilterToTestTFM { get; set; }
+
+        [Required]
+        public string FilterToOSGroup { get; set; }
+
+        [Output]
+        public ITaskItem[] FilteredUploadItems
+        {
+            get { return _copyLocalItems.ToArray(); }
+        }
+
+        private readonly List<ITaskItem> _copyLocalItems = new List<ITaskItem>();
+
+        public override bool Execute()
+        {
+
+            Debugger.Launch();
+            string[] filesToSearch = Directory.GetFiles(TestOutputDirListsLocation);
```

Comment:
you never use filesToSearch again, can just put the GetFiles call in the foreach.